modules/library/coverage/src/main/java/org/geotools/coverage/grid/io/AbstractGridCoverage2DReader.java

```diff
@@ -1079,7 +1084,12 @@ public int getNumOverviews(String coverageName) {
             throw new IllegalArgumentException("The specified coverageName " + coverageName
                     + "is not supported");
         }
-        return overViewResolutions != null ? overViewResolutions.length : 0;
+        if (dtLayout == null) {
+            // Back to the default
+            return numOverviews;
+        }
+        return dtLayout.getNumInternalOverviews()
```

Comment:
Seeing this code around repeated, makes sense to have a method in DatasetLayout instead? getNumOverviews() implying it's the total one?